common/utils.js

```diff
@@ -63,6 +66,33 @@ export class Encoding {
         return Encoding.makeWebsafe(fullKey.toString('base64'));
     }
 
+    /**
+     * Decodes array of keys from provided text.
+     * @param {string} key 
+     * @returns {string[]}
+     */
+    static fromShortKeyArray(key) {
+        const binData = Buffer.from(key, 'base64');
+        const result = [];
+        let offset = 0;
+
+        while (offset < binData.length) {
+            const size = binData.length - offset;
+
+            if (size < kElementIdSize) {
+                break;
+            }
+            var elementBuff = Buffer.alloc(kElementIdWithFlagsSize);
```

Comment:
Would be more efficient to create a buffer once and reuse in iterations.

Reply:
Fixed